src/watch.ts

```diff
@@ -70,6 +77,10 @@ export class Watch {
             errOut = err;
             done(err);
         });
+        // TODO: I don't love this, because both 'error' and 'close' call the done handler with the same error
+        // We should probably only do one or the other, but there's challenges because of async delivery and it's
+        // important to know if the close event is occurring because of an error. So for now, this needs to be
+        // handled in the client.
```

Comment:
Instead of leaving the code in watcher as it is and having workarounds for all potential users of the watcher that are IMHO unreliable anyway, I think it would have been better to fix it here. Would a boolean variable indicating that the done callback was called be sufficient? That way we could guarantee that the done callback is called at most once, that is exactly what users of the watcher expect.